algosdk/atomic_transaction_composer.py

```diff
@@ -317,51 +332,53 @@ def execute(self, client, wait_rounds):
             )
 
         self.submit(client)
-        transaction.wait_for_confirmation(client, self.tx_ids[0], wait_rounds)
+        resp = transaction.wait_for_confirmation(
+            client, self.tx_ids[0], wait_rounds
+        )
         self.status = AtomicTransactionComposerStatus.COMMITTED
 
-        confirmed_round = -1
+        confirmed_round = resp["confirmed-round"]
         method_results = []
 
         for i, tx_id in enumerate(self.tx_ids):
+            raw_value = None
+            return_value = None
+            decode_error = None
+
             if i not in self.method_dict:
                 continue
             # Return is void
             if self.method_dict[i].returns is None:
```

Comment:
Is this check still right?

Reply:
Good catch, this should be changed as `returns` now has `void` and is always required. I will also strengthen the unit tests in the next commit.